### ISSUE TITLE
Sampling screen - 'Single sample' mean indicator and timing

### ISSUE THREAD

**Issue by matthew-blackman** [phetsims/projectile-data-lab#17]
We are working on fine-tuning the timing of the sampling and mean indicator for the 'Single sample' mode of the Sampling screen.

The following patch works to get the mean indicator to appear with a delay after each sample, but crashes when the sim is reset. 

We would like to get it working so that the mean indicator correctly updates when creating samples, changing samples, or changing fields.

<details>

```diff

Subject: [PATCH] Update sample mean display timing
---
Index: js/sampling/view/SamplingCanvasNode.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/js/sampling/view/SamplingCanvasNode.ts b/js/sampling/view/SamplingCanvasNode.ts
--- a/js/sampling/view/SamplingCanvasNode.ts	(revision dc346f0707bd23d92f60a324eed244a5c80b8794)
+++ b/js/sampling/view/SamplingCanvasNode.ts	(date 1702671426744)
@@ -55,29 +55,6 @@
       // wish to show the undarkened ones while flying through the air, or otherwise try to match the design from screens 1-3
       this.drawProjectile( context, projectiles[ i ], true, true );
     }
-
-    // Draw the mean marker (if the sample is complete)
-    if ( projectiles.length === this.fieldProperty.value.sampleSize ) {
-      const meanX = _.mean( projectiles.map( projectile => projectile.x ) );
-      const viewPoint = this.modelViewTransform.modelToViewXY( meanX, 0 );
-
-      const SIDE_LENGTH = 20; // Define the side length of the triangle
-      const height = ( Math.sqrt( 3 ) / 2 ) * SIDE_LENGTH; // Calculate the height of the equilateral triangle
-
-      // Draw a purple triangle pointing down at that viewPoint which has {x:number,y:number}
-      context.beginPath();
-      context.moveTo( viewPoint.x, viewPoint.y ); // Move to the top vertex of the triangle
-      context.lineTo( viewPoint.x - SIDE_LENGTH / 2, viewPoint.y - height ); // Line to bottom-left vertex
-      context.lineTo( viewPoint.x + SIDE_LENGTH / 2, viewPoint.y - height ); // Line to bottom-right vertex
-      context.closePath();
-      context.fillStyle = PDLColors.meanMarkerFillProperty.value.toCSS();
-      context.fill();
-
-      // And add a black stroke around it
-      context.lineWidth = 1;
-      context.strokeStyle = PDLColors.meanMarkerStrokeProperty.value.toCSS();
-      context.stroke();
-    }
   }
 }
 
Index: js/sampling/model/SamplingField.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/js/sampling/model/SamplingField.ts b/js/sampling/model/SamplingField.ts
--- a/js/sampling/model/SamplingField.ts	(revision dc346f0707bd23d92f60a324eed244a5c80b8794)
+++ b/js/sampling/model/SamplingField.ts	(date 1702671426743)
@@ -8,6 +8,8 @@
 import PDLEventTimer from '../../common/model/PDLEventTimer.js';
 import HistogramData from '../../common/model/HistogramData.js';
 import Tandem from '../../../../tandem/js/Tandem.js';
+import Property from '../../../../axon/js/Property.js';
+import Multilink from '../../../../axon/js/Multilink.js';
 
 /**
  * The SamplingField is an extension of the Field class that adds fields for the Sampling model. Note in order to support
@@ -26,11 +28,16 @@
 // can still see data sweep in when in single mode.
 const WITHIN_SAMPLE_TIME_CONTINUOUS = 0;
 
+// This is the delay between the last projectile landing and the mean symbol appearing, in 'Single sample' mode.
+const MEAN_SYMBOL_DELAY_TIME = 0.3;
+const NEXT_SAMPLE_DELAY_TIME = 0.5;
+
 export default class SamplingField extends Field {
   public override identifier: string;
 
   private readonly withinSampleTimer: PDLEventTimer;
-  private readonly betweenSamplesTimer = new PDLEventTimer( 0.2 );
+  private readonly betweenSamplesTimer = new PDLEventTimer( MEAN_SYMBOL_DELAY_TIME + NEXT_SAMPLE_DELAY_TIME );
+  private meanIndicatorDelayTimer: PDLEventTimer | null = null;
 
   // The simulation begins with 0 samples, showing sample 0 of 0. When a sample begins, this number increases. This is
   // a convenience Property to simplify the user interface -- it is fully computable from the Projectiles.
@@ -40,6 +47,11 @@
   // This is a convenience Property to simplify the user interface -- it is fully computable from the Projectiles.
   public readonly numberOfCompletedSamplesProperty: NumberProperty;
 
+  // This property is used to set the visibility and position of the mean indicator.
+  // If it is null, the current sample is not yet complete and the mean indicator is not visible.
+  // Note: In 'Single sample' mode, there is a delay between the last projectile in a sample and the mean indicator appearing.
+  public readonly sampleMeanProperty: Property<number | null>;
+
   // private currentLandedCount = 0;
   private readonly singleModeWithinSamplePeriod: number;
 
@@ -58,14 +70,18 @@
 
     // Increase the total time as the sample size increases, so that larger samples take longer but not too long.
     const totalSampleTime =
-      this.sampleSize === 2 ? 0.5 :
-      this.sampleSize === 5 ? 0.75 :
-      this.sampleSize === 15 ? 0.85 :
-      this.sampleSize === 40 ? 1 :
+      this.sampleSize === 2 ? 0.7 :
+      this.sampleSize === 5 ? 1.2 :
+      this.sampleSize === 15 ? 2 :
+      this.sampleSize === 40 ? 3 :
       0;
 
     assert && assert( totalSampleTime > 0, 'totalSampleTime should be greater than 0' );
 
+    this.sampleMeanProperty = new Property<number | null>( null, {
+      reentrant: true
+    } );
+
     this.singleModeWithinSamplePeriod = totalSampleTime / this.sampleSize;
 
     this.withinSampleTimer = new PDLEventTimer( this.singleModeWithinSamplePeriod );
@@ -73,6 +89,12 @@
     this.presetLauncherProperty.value = launcher;
     this.updateCounts();
 
+    Multilink.multilink( [ this.selectedSampleProperty, this.sampleMeanProperty ], ( selectedSample, sampleMean ) => {
+      if ( sampleMean !== null ) {
+        this.sampleMeanProperty.value = _.mean( this.getProjectilesInSelectedSample().map( projectile => projectile.x ) );
+      }
+    } );
+
     Tandem.PHET_IO_ENABLED && phet.phetio.phetioEngine.phetioStateEngine.stateSetEmitter.addListener( () => {
       this.updateCounts();
     } );
@@ -123,6 +145,9 @@
 
   public startNewSample(): void {
 
+    // When starting a new sample, the mean indicator should be hidden.
+    this.sampleMeanProperty.value = null;
+
     // Any time we start a new sample, if an old sample was in-progress, we must complete it immediately/synchronously
     // before starting the new one. Typically, this would only happen when the user clicks the launch button while
     // continuous launching is enabled.
@@ -136,7 +161,7 @@
     updated && this.updateCounts();
 
     // Create the first projectile in the new sample.
-    this.createLandedProjectile();
+    // this.createLandedProjectile();
 
     // Show the new sample in the selector panel. Note this means if sample 2/8 was selected, then we
     // create a new one, it will jump to 9/9. This is the desired behavior.
@@ -149,6 +174,12 @@
   }
 
   public step( dt: number, isContinuousLaunching: boolean ): void {
+    if ( this.meanIndicatorDelayTimer !== null ) {
+      this.meanIndicatorDelayTimer.step( dt, () => {
+        this.sampleMeanProperty.value = _.mean( this.getProjectilesInSelectedSample().map( projectile => projectile.x ) );
+        this.meanIndicatorDelayTimer = null;
+      } );
+    }
 
     if ( isContinuousLaunching ) {
       this.betweenSamplesTimer.step( dt, () => {
@@ -163,6 +194,10 @@
 
       if ( this.getTotalProjectileCount() % this.sampleSize === 0 ) {
 
+        // When a sample is completed, timeSinceSampleFinished will be set from null to 0
+        this.meanIndicatorDelayTimer = new PDLEventTimer( MEAN_SYMBOL_DELAY_TIME );
+        this.meanIndicatorDelayTimer.restart();
+
         // Finished a sample, schedule the next one to begin soon
         this.withinSampleTimer.stop();
         this.betweenSamplesTimer.restart();
@@ -179,8 +214,11 @@
 
     this.withinSampleTimer.stop();
     this.betweenSamplesTimer.stop();
+    this.meanIndicatorDelayTimer = null;
 
     this.updateCounts();
+
+    this.sampleMeanProperty.reset();
   }
 
   public setLaunchMode( launchMode: 'single' | 'continuous' ): void {
Index: js/sampling/view/SamplingScreenView.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/js/sampling/view/SamplingScreenView.ts b/js/sampling/view/SamplingScreenView.ts
--- a/js/sampling/view/SamplingScreenView.ts	(revision dc346f0707bd23d92f60a324eed244a5c80b8794)
+++ b/js/sampling/view/SamplingScreenView.ts	(date 1702671426745)
@@ -19,6 +19,8 @@
 import LauncherNode from '../../common/view/LauncherNode.js';
 import SamplingCanvasNode from './SamplingCanvasNode.js';
 import ProjectileDataLabStrings from '../../ProjectileDataLabStrings.js';
+import MeanIndicatorNode from '../../common/view/MeanIndicatorNode.js';
+import DerivedProperty from '../../../../axon/js/DerivedProperty.js';
 
 type SelfOptions = EmptySelfOptions;
 
@@ -43,6 +45,21 @@
       } );
     this.projectileCanvasLayer.addChild( projectileCanvas );
 
+    const meanIndicatorVisibleProperty = new DerivedProperty( [ model.sampleMeanProperty ],
+      sampleMean => sampleMean !== null );
+
+    const meanIndicatorNode = new MeanIndicatorNode( 14, {
+      visibleProperty: meanIndicatorVisibleProperty,
+      bottom: this.modelViewTransform.modelToViewY( 0 )
+    } );
+    this.projectileCanvasLayer.addChild( meanIndicatorNode );
+
+    model.sampleMeanProperty.link( mean => {
+      if ( mean !== null ) {
+        meanIndicatorNode.centerX = this.modelViewTransform.modelToViewX( mean );
+      }
+    } );
+
     this.launcherNode = new LauncherNode(
       this.modelViewTransform,
       model.meanLaunchAngleProperty,
Index: js/sampling/model/SamplingModel.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/js/sampling/model/SamplingModel.ts b/js/sampling/model/SamplingModel.ts
--- a/js/sampling/model/SamplingModel.ts	(revision dc346f0707bd23d92f60a324eed244a5c80b8794)
+++ b/js/sampling/model/SamplingModel.ts	(date 1702671426744)
@@ -28,11 +28,12 @@
 export default class SamplingModel extends PDLModel<SamplingField> {
 
   public readonly sampleSizeProperty: Property<number>;
+  public readonly presetLauncherProperty: NumberProperty;
 
   public readonly numberOfStartedSamplesProperty: DynamicProperty<number, number, SamplingField>;
   public readonly selectedSampleProperty: DynamicProperty<number, number, SamplingField>;
   public readonly numberOfCompletedSamplesProperty: DynamicProperty<number, number, SamplingField>;
-  public readonly presetLauncherProperty: NumberProperty;
+  public readonly sampleMeanProperty: DynamicProperty<number | null, number | null, SamplingField>;
 
   public constructor( providedOptions: SamplingModelOptions ) {
 
@@ -85,6 +86,10 @@
       derive: t => t.numberOfCompletedSamplesProperty
     } );
 
+    this.sampleMeanProperty = new DynamicProperty<number | null, number | null, SamplingField>( this.fieldProperty, {
+      derive: t => t.sampleMeanProperty
+    } );
+
     // When the launch mode changes, update the timing between projectiles within a sample.
     this.launchModeProperty.link( launchMode => {
       this.fields.forEach( field => field.setLaunchMode( launchMode ) );

**Comment by samreid** [comment]
I fixed the bugs above and addressed some other problems, but wasn't quite ready to commit.  Here is the status:

<details>

```diff

Subject: [PATCH] Move field panel above projectile selector panel, see https://github.com/phetsims/projectile-data-lab/issues/7
---
Index: js/sampling/view/SamplingCanvasNode.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/js/sampling/view/SamplingCanvasNode.ts b/js/sampling/view/SamplingCanvasNode.ts
--- a/js/sampling/view/SamplingCanvasNode.ts	(revision dc346f0707bd23d92f60a324eed244a5c80b8794)
+++ b/js/sampling/view/SamplingCanvasNode.ts	(date 1702672986148)
@@ -55,29 +55,6 @@
       // wish to show the undarkened ones while flying through the air, or otherwise try to match the design from screens 1-3
       this.drawProjectile( context, projectiles[ i ], true, true );
     }
-
-    // Draw the mean marker (if the sample is complete)
-    if ( projectiles.length === this.fieldProperty.value.sampleSize ) {
-      const meanX = _.mean( projectiles.map( projectile => projectile.x ) );
-      const viewPoint = this.modelViewTransform.modelToViewXY( meanX, 0 );
-
-      const SIDE_LENGTH = 20; // Define the side length of the triangle
-      const height = ( Math.sqrt( 3 ) / 2 ) * SIDE_LENGTH; // Calculate the height of the equilateral triangle
-
-      // Draw a purple triangle pointing down at that viewPoint which has {x:number,y:number}
-      context.beginPath();
-      context.moveTo( viewPoint.x, viewPoint.y ); // Move to the top vertex of the triangle
-      context.lineTo( viewPoint.x - SIDE_LENGTH / 2, viewPoint.y - height ); // Line to bottom-left vertex
-      context.lineTo( viewPoint.x + SIDE_LENGTH / 2, viewPoint.y - height ); // Line to bottom-right vertex
-      context.closePath();
-      context.fillStyle = PDLColors.meanMarkerFillProperty.value.toCSS();
-      context.fill();
-
-      // And add a black stroke around it
-      context.lineWidth = 1;
-      context.strokeStyle = PDLColors.meanMarkerStrokeProperty.value.toCSS();
-      context.stroke();
-    }
   }
 }
 
Index: js/sampling/model/SamplingField.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/js/sampling/model/SamplingField.ts b/js/sampling/model/SamplingField.ts
--- a/js/sampling/model/SamplingField.ts	(revision dc346f0707bd23d92f60a324eed244a5c80b8794)
+++ b/js/sampling/model/SamplingField.ts	(date 1702673848908)
@@ -8,6 +8,7 @@
 import PDLEventTimer from '../../common/model/PDLEventTimer.js';
 import HistogramData from '../../common/model/HistogramData.js';
 import Tandem from '../../../../tandem/js/Tandem.js';
+import Property from '../../../../axon/js/Property.js';
 
 /**
  * The SamplingField is an extension of the Field class that adds fields for the Sampling model. Note in order to support
@@ -26,11 +27,18 @@
 // can still see data sweep in when in single mode.
 const WITHIN_SAMPLE_TIME_CONTINUOUS = 0;
 
+// This is the delay between the last projectile landing and the mean symbol appearing, in 'Single sample' mode.
+const MEAN_SYMBOL_DELAY_TIME = 0.3;
+const NEXT_SAMPLE_DELAY_TIME = 0.5;
+
 export default class SamplingField extends Field {
   public override identifier: string;
 
   private readonly withinSampleTimer: PDLEventTimer;
-  private readonly betweenSamplesTimer = new PDLEventTimer( 0.2 );
+  private readonly betweenSamplesTimer = new PDLEventTimer( MEAN_SYMBOL_DELAY_TIME + NEXT_SAMPLE_DELAY_TIME );
+
+  // After a sample completes, this timer counts down until we show the mean indicator.
+  private meanIndicatorDelayTimer: PDLEventTimer | null = null;
 
   // The simulation begins with 0 samples, showing sample 0 of 0. When a sample begins, this number increases. This is
   // a convenience Property to simplify the user interface -- it is fully computable from the Projectiles.
@@ -40,6 +48,11 @@
   // This is a convenience Property to simplify the user interface -- it is fully computable from the Projectiles.
   public readonly numberOfCompletedSamplesProperty: NumberProperty;
 
+  // This property is used to set the visibility and position of the mean indicator.
+  // If it is null, the current sample is not yet complete and the mean indicator is not visible.
+  // Note: In 'Single sample' mode, there is a delay between the last projectile in a sample and the mean indicator appearing.
+  public readonly sampleMeanProperty: Property<number | null>;
+
   // private currentLandedCount = 0;
   private readonly singleModeWithinSamplePeriod: number;
 
@@ -58,14 +71,18 @@
 
     // Increase the total time as the sample size increases, so that larger samples take longer but not too long.
     const totalSampleTime =
-      this.sampleSize === 2 ? 0.5 :
-      this.sampleSize === 5 ? 0.75 :
-      this.sampleSize === 15 ? 0.85 :
-      this.sampleSize === 40 ? 1 :
+      this.sampleSize === 2 ? 0.7 :
+      this.sampleSize === 5 ? 1.2 :
+      this.sampleSize === 15 ? 2 :
+      this.sampleSize === 40 ? 3 :
       0;
 
     assert && assert( totalSampleTime > 0, 'totalSampleTime should be greater than 0' );
 
+    this.sampleMeanProperty = new Property<number | null>( null, {
+      reentrant: true
+    } );
+
     this.singleModeWithinSamplePeriod = totalSampleTime / this.sampleSize;
 
     this.withinSampleTimer = new PDLEventTimer( this.singleModeWithinSamplePeriod );
@@ -73,6 +90,20 @@
     this.presetLauncherProperty.value = launcher;
     this.updateCounts();
 
+    this.selectedSampleProperty.lazyLink( () => {
+
+      // TODO: Factor out: https://github.com/phetsims/projectile-data-lab/issues/17
+      const projectilesInSelectedSample = this.getProjectilesInSelectedSample();
+
+      // TODO: This wouldn't work if we ever have sample size = 1, see https://github.com/phetsims/projectile-data-lab/issues/17
+      if ( projectilesInSelectedSample.length > 1 ) {
+        this.sampleMeanProperty.value = _.mean( projectilesInSelectedSample.map( projectile => projectile.x ) );
+      }
+      else {
+        this.sampleMeanProperty.value = null;
+      }
+    } );
+
     Tandem.PHET_IO_ENABLED && phet.phetio.phetioEngine.phetioStateEngine.stateSetEmitter.addListener( () => {
       this.updateCounts();
     } );
@@ -123,6 +154,9 @@
 
   public startNewSample(): void {
 
+    // When starting a new sample, the mean indicator should be hidden.
+    this.sampleMeanProperty.value = null;
+
     // Any time we start a new sample, if an old sample was in-progress, we must complete it immediately/synchronously
     // before starting the new one. Typically, this would only happen when the user clicks the launch button while
     // continuous launching is enabled.
@@ -135,8 +169,11 @@
     // Make sure we are pointing at the right sample for the new Projectile
     updated && this.updateCounts();
 
-    // Create the first projectile in the new sample.
-    this.createLandedProjectile();
+    // Create the first projectile in the new sample if there is none, otherwise immediately clear and schedule one
+    // to appear when the timer expires.
+    if ( this.getTotalProjectileCount() === 0 ) {
+      this.createLandedProjectile();
+    }
 
     // Show the new sample in the selector panel. Note this means if sample 2/8 was selected, then we
     // create a new one, it will jump to 9/9. This is the desired behavior.
@@ -149,6 +186,18 @@
   }
 
   public step( dt: number, isContinuousLaunching: boolean ): void {
+    if ( this.meanIndicatorDelayTimer !== null ) {
+      this.meanIndicatorDelayTimer.step( dt, () => {
+        const projectilesInSelectedSample = this.getProjectilesInSelectedSample();
+        if ( projectilesInSelectedSample.length > 0 ) {
+          this.sampleMeanProperty.value = _.mean( projectilesInSelectedSample.map( projectile => projectile.x ) );
+        }
+        else {
+          this.sampleMeanProperty.value = null;
+        }
+        this.meanIndicatorDelayTimer = null;
+      } );
+    }
 
     if ( isContinuousLaunching ) {
       this.betweenSamplesTimer.step( dt, () => {
@@ -163,6 +212,10 @@
 
       if ( this.getTotalProjectileCount() % this.sampleSize === 0 ) {
 
+        // When a sample is completed, timeSinceSampleFinished will be set from null to 0
+        this.meanIndicatorDelayTimer = new PDLEventTimer( MEAN_SYMBOL_DELAY_TIME );
+        this.meanIndicatorDelayTimer.restart();
+
         // Finished a sample, schedule the next one to begin soon
         this.withinSampleTimer.stop();
         this.betweenSamplesTimer.restart();
@@ -181,6 +234,9 @@
     this.betweenSamplesTimer.stop();
 
     this.updateCounts();
+
+    this.sampleMeanProperty.reset();
+    this.meanIndicatorDelayTimer = null;
   }
 
   public setLaunchMode( launchMode: 'single' | 'continuous' ): void {
Index: js/sampling/model/SamplingModel.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/js/sampling/model/SamplingModel.ts b/js/sampling/model/SamplingModel.ts
--- a/js/sampling/model/SamplingModel.ts	(revision dc346f0707bd23d92f60a324eed244a5c80b8794)
+++ b/js/sampling/model/SamplingModel.ts	(date 1702672986143)
@@ -28,11 +28,12 @@
 export default class SamplingModel extends PDLModel<SamplingField> {
 
   public readonly sampleSizeProperty: Property<number>;
+  public readonly presetLauncherProperty: NumberProperty;
 
   public readonly numberOfStartedSamplesProperty: DynamicProperty<number, number, SamplingField>;
   public readonly selectedSampleProperty: DynamicProperty<number, number, SamplingField>;
   public readonly numberOfCompletedSamplesProperty: DynamicProperty<number, number, SamplingField>;
-  public readonly presetLauncherProperty: NumberProperty;
+  public readonly sampleMeanProperty: DynamicProperty<number | null, number | null, SamplingField>;
 
   public constructor( providedOptions: SamplingModelOptions ) {
 
@@ -85,6 +86,10 @@
       derive: t => t.numberOfCompletedSamplesProperty
     } );
 
+    this.sampleMeanProperty = new DynamicProperty<number | null, number | null, SamplingField>( this.fieldProperty, {
+      derive: t => t.sampleMeanProperty
+    } );
+
     // When the launch mode changes, update the timing between projectiles within a sample.
     this.launchModeProperty.link( launchMode => {
       this.fields.forEach( field => field.setLaunchMode( launchMode ) );
Index: js/sampling/view/SamplingScreenView.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/js/sampling/view/SamplingScreenView.ts b/js/sampling/view/SamplingScreenView.ts
--- a/js/sampling/view/SamplingScreenView.ts	(revision dc346f0707bd23d92f60a324eed244a5c80b8794)
+++ b/js/sampling/view/SamplingScreenView.ts	(date 1702672986153)
@@ -19,6 +19,8 @@
 import LauncherNode from '../../common/view/LauncherNode.js';
 import SamplingCanvasNode from './SamplingCanvasNode.js';
 import ProjectileDataLabStrings from '../../ProjectileDataLabStrings.js';
+import MeanIndicatorNode from '../../common/view/MeanIndicatorNode.js';
+import DerivedProperty from '../../../../axon/js/DerivedProperty.js';
 
 type SelfOptions = EmptySelfOptions;
 
@@ -43,6 +45,21 @@
       } );
     this.projectileCanvasLayer.addChild( projectileCanvas );
 
+    const meanIndicatorVisibleProperty = new DerivedProperty( [ model.sampleMeanProperty ],
+      sampleMean => sampleMean !== null );
+
+    const meanIndicatorNode = new MeanIndicatorNode( 14, {
+      visibleProperty: meanIndicatorVisibleProperty,
+      bottom: this.modelViewTransform.modelToViewY( 0 )
+    } );
+    this.projectileCanvasLayer.addChild( meanIndicatorNode );
+
+    model.sampleMeanProperty.link( mean => {
+      if ( mean !== null ) {
+        meanIndicatorNode.centerX = this.modelViewTransform.modelToViewX( mean );
+      }
+    } );
+
     this.launcherNode = new LauncherNode(
       this.modelViewTransform,
       model.meanLaunchAngleProperty,

```

</details>

**Comment by samreid** [comment]
OK after I verified some behavior, I added some TODOs and will commit. Will need help from @matthew-blackman cleaning up and getting it production worthy.

**Comment by samreid** [comment]
9 TODOs remain for this issue

**Comment by samreid** [comment]
On hold until we get design feedback on Friday.

**Comment by matthew-blackman** [comment]
The behavior for Single sample mode is working mostly correctly. We will track further bugs and edge cases in https://github.com/phetsims/projectile-data-lab/issues/22. Closing.
